storage/rocksdb/rdb_datadic.cc

```diff
@@ -925,13 +988,67 @@ uint Rdb_key_def::pack_record(const TABLE *const tbl, uchar *const pack_buffer,
     tuple = pack_field(field, &m_pack_info[i], tuple, packed_tuple, pack_buffer,
                        unpack_info, n_null_fields);
 
+    // Check if this column is covered.
+    if (unpack_info && m_index_type == INDEX_TYPE_SECONDARY) {
+      size_t data_length = field->data_length();
+      uint16 key_length;
+
+      // We always need the PK as a suffix, but in some cases MySQL won't add
+      // the PK suffix to a secondary key, and therefore it won't be included
+      // in key_info. If so, we need to get the key length for key parts from
+      // the PK directly.
+      DBUG_ASSERT(tbl->key_info != nullptr);
+      if (i >= tbl->key_info[get_keyno()].actual_key_parts) {
```

Comment:
It looks like your code covers this now, but let's add a test case for it. Create a table with 20 varchar(64) columns, create a primary key with 4 of the varchar columns using varchar(8), then a secondary key with the rest of the 20 columns using varchar(8). Each key column only uses the first 8 characters of varchar. Then have various rows with varchar strings of < 8 bytes, or > 8 bytes for the primary and secondary columns, and select on those columns to see if the pk is used for covered index or not (should always fail).

i.e.

```
CREATE TABLE T1 (
  P1 VARCHAR(64), P2 VARCHAR(64), P3 VARCHAR(64), P4 VARCHAR(64),
  S1 VARCHAR(64), S2 VARCHAR(64), ..., S16 VARCHAR(64),
  PRIMARY KEY (P1, P2, P3, P4),
  KEY (S1, S2, S3, ..., S16) 
) ENGINE=InnoDB;

INSERT INTO t1 values ('1', '2', '3', '4', '5', '6, '7', ..., '20');
SELECT P4, S2 FROM t1 where S1 = '5'; # Should fail to return P4 as covered because there were not enough bits to cover it.
```